### ISSUE TITLE
Minified File?

### ISSUE THREAD

**Issue by danzen** [CurriculumAssociates/canvas-latex#1]
Is there a single minified file we can use with CreateJS?  If it is supposed to come from some node build, could you outline the steps - and where I am supposed to find it once we run those node commands.  Thanks.  Looking forward to trying it out!

**Comment by ry-randall** [comment]
@danzen I just pushed a change which should allow you to do what you need.

Here are the steps you'll want to follow:

1. Pull the latest
2. Run `npm install`
3. Run `npm build`
4. Run `open dist/index.html`

Note: The minified file is called `canvas-latex.js` and should live in the dist directory after you run build.

Let me know if you hit issues.

**Comment by danzen** [comment]
Well... I grabbed the latest and unzipped.  Then opened the Node.js command prompt and ran npm install.  It went for a while and then gave me this:

D:\canvas-latex-master\node_modules\canvas>if not defined npm_config_node_gyp (node "C:\Users\danzen\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Users\danzen\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
gyp ERR! configure error
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at Object.failNoPython (C:\Users\danzen\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\configure.js:484:19)
gyp ERR! stack     at Object.<anonymous> (C:\Users\danzen\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\configure.js:509:16)
gyp ERR! stack     at C:\Users\danzen\AppData\Roaming\npm\node_modules\npm\node_modules\graceful-fs\polyfills.js:284:29
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:123:15)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\danzen\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd D:\canvas-latex-master\node_modules\canvas
gyp ERR! node -v v6.10.2
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! canvas@1.3.16 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the canvas@1.3.16 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

-------------

I tried the npm build after and it gave me this:
npm WARN build `npm build` called with no arguments. Did you mean to `npm run-script build`?

Sorry if it is just my nodes setup or something... I have run other things successfully.  We can try and figure this out - perhaps in parallel, you could post a version of canvas-latex.js and I can try that out too.  Thanks!

**Comment by ry-randall** [comment]
Thanks for digging into this @danzen. The `canvas` dependency has been removed now. Would you try pulling the latest and running:
`npm install`
`npm run build`

Let me know if you still hit issues.

**Comment by danzen** [comment]
Tried again and got the message below.  I am in Windows so is it copy and not cp?  Anyway - I tried copy and it said:
D:\canvas-latex-master>copy -a demo/ dist/
The system cannot find the file specified.
despite there being a demo directory in the folder.  So, I am back here... if you want, you could upload a version of the canvas-latex.js and I can try it out in the meantime.

Here is the original message - sorry for the bother!

D:\canvas-latex-master>npm run build

> @curriculumassociates/canvas-latex@0.9.0 prebuild D:\canvas-latex-master
> npm run copy-demo

> @curriculumassociates/canvas-latex@0.9.0 copy-demo D:\canvas-latex-master
> cp -a demo/ dist/

'cp' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @curriculumassociates/canvas-latex@0.9.0 copy-demo: `cp -a demo/ dist/`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @curriculumassociates/canvas-latex@0.9.0 copy-demo script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\danzen\AppData\Roaming\npm-cache\_logs\2018-10-02T04_28_53_770Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @curriculumassociates/canvas-latex@0.9.0 prebuild: `npm run copy-demo`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @curriculumassociates/canvas-latex@0.9.0 prebuild script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\danzen\AppData\Roaming\npm-cache\_logs\2018-10-02T04_28_53_806Z-debug.log

D:\canvas-latex-master>

**Comment by ry-randall** [comment]
Hey Dan. I have a PR open to resolve the copy issue. Thanks for finding that. 

In the meantime, feel free to play around:

[canvas-latex.js.zip](https://github.com/CurriculumAssociates/canvas-latex/files/2438347/canvas-latex.js.zip)

**Comment by danzen** [comment]
Thanks folks - it worked this time.  Here is CanvasLatex working in ZIM - which is powered by CreateJS to give conveniences, components and controls - https://zimjs.com and here is the demo:
https://zimjs.com/explore/math.html - you would be welcome to add ZIM to your list of supported renderers.  Thanks for the helpful library!

**Comment by ry-randall** [comment]
@danzen This site is really cool! Looks like a great way to teach kids how to program interactively.

**Comment by danzen** [comment]
Thanks Ryan!  I have adjusted the example to include the fonts - ZIM has a way of loading fonts too which points through to the CreateJS PreloadJS and its FontLoader.  I had it commented out before as I did not notice much difference - but perhaps there is for certain characters.  Anyway, if you take a look a the example, the fonts are loaded now.  Thanks again - and it is good to show our people Math support.  There are a number of folks around the world working on e-learning apps in ZIM so this helps.
